src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/pack.py

```diff
@@ -0,0 +1,106 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import base64
+from knack.log import get_logger
+from knack.util import CLIError
+from azure.cli.core.commands import LongRunningOperation
+
+from ._constants import ORYX_PACK_BUILDER_IMAGE
+from ._stream_utils import stream_logs
+from ._utils import (
+    validate_managed_registry,
+    get_validate_platform,
+    get_custom_registry_credentials
+)
+from ._client_factory import cf_acr_registries
+from .run import prepare_source_location
+
+PACK_NOT_SUPPORTED = 'Pack is only available for managed registries.'
+PACK_TASK_YAML_FMT = '''version: v1.0.0
+steps:
+  - cmd: mcr.microsoft.com/oryx/pack:stable build {image_name} --builder {builder} {no_pull} --env REGISTRY_NAME={{{{.Run.Registry}}}} -p .
+    timeout: 28800
+  - push: ["{image_name}"]
+    timeout: 1800
+'''
+
+logger = get_logger(__name__)
+
+
+def acr_pack(cmd,  # pylint: disable=too-many-locals
+             client,
+             registry_name,
+             image_name,
+             source_location,
+             builder=ORYX_PACK_BUILDER_IMAGE,
+             pull=False,
+             no_format=False,
+             no_logs=False,
+             no_wait=False,
+             timeout=None,
+             resource_group_name=None,
+             platform=None,
+             auth_mode=None):
+
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, PACK_NOT_SUPPORTED)
```

Comment:
Can you replace it with 

registry, resource_group_name = get_registry_by_name(cmd.cli_ctx, registry_name)
login_server = registry.login_server

Reply:
don't we also want to ensure that the registry is a managed one?